### PR TITLE
[G2M] Modules/login

### PR DESCRIPTION
**Pull request by hyx131**
**Changes:**
- add `Login` module

**Screenshots:**
Screen Shot 2022-03-28 at 12 11 43 PM
Screen Shot 2022-03-28 at 12 11 25 PM

### CODE REVIEW

**Comment by sallkall** [comment]
@hyx131 for the input check, it should have 2 error stages:
1. "Required" (empty field error)
image

2. "Invalid" (incorrect formating error)
image

^Using some other website's UI as examples

**Comment by hyx131** [comment]
@sallkall they are there, one is set by `Login` (empty email error), the other comes default from our `Textfield` component (incorrect email format)

Screen Shot 2022-03-28 at 12 51 58 PM

Screen Shot 2022-03-28 at 12 52 05 PM

**Comment by sallkall** [comment]
> @sallkall they are there, one is set by `Login` (empty email error), the other comes default from our `Textfield` component (incorrect email format)
> 
> 
> 
> Screen Shot 2022-03-28 at 12 51 58 PM
> 
> 
> 
> Screen Shot 2022-03-28 at 12 52 05 PM
> 
> 
> 
> 
> 
> 

Ohhh weird, I didn't see this when I tired. That's great thanks for adding it in 🤩

**Comment by sallkall** [comment]
Oh i see, this tooltip warning only appears on enter - should we just have it always until we detect the correct syntax? 

https://user-images.githubusercontent.com/33563070/160448927-97db6ea9-8244-4950-9fba-40b6747057d0.mov

**Comment by sallkall** [comment]
@vlatawiec ^ what do you think? Did you have anything in mind for this input feedback part? 😄

**Comment by hyx131** [comment]
> Oh i see, this tooltip warning only appears on enter - should we just have it always until we detect the correct syntax?
> 
>  Screen.Recording.2022-03-28.at.12.48.33.PM.mov 
> ^like how it only goes back to normal state when they detect `@` and a `.` after?

yup, we can do that too

**Comment by sallkall** [comment]
> Oh i see, this tooltip warning only appears on enter - should we just have it always until we detect the correct syntax?
> 
>  Screen.Recording.2022-03-28.at.12.48.33.PM.mov 
> ^like how it only goes back to normal state when they detect `@` and a `.` after?

I don't think it'll make them feel 'bad' cuz we're just trying to give them feedback and guide them to make a correct input. On enter is also fine but then they have to 'enter' (extra step) to get feedback. I'm good with either will wait for @vlatawiec 's input as she may have thought more about this than me :D

File: src/modules/login.js [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+    if (field === 'passcode') {
+      setPasscode(value)
+      const match = (value).match(/[^A-F\d]/gi)
+
+      if (value && errors.passcodeError) {
+        setErrors((prev) => ({ ...prev, passcodeError: '' }))
+      }
+      if (!value || match) {
+        setErrors((prev) => ({ ...prev, passcodeError: match ? `'${[...match]}' is not a valid character for passcode` : 'Passcode required' }))
+        changes = { ...changes, errorConfig: { ...errors, passcodeError: match ? `'${[...match]}' is not a valid character for passcode` : 'Passcode required' } }
```
Comment (geoerika) [inline src/modules/login.js:160]: Just a remarque... had to hide the new file tree in github to see this line. Can we perhaps agree to limit the line length to some more reasonable reading format? This one stands at 154 characters. The somewhat standard length would be 100-120.

**Review by vlatawiec — APPROVED** [review]
Looks great! For the error input field, I think it would be ideal to have the message appear immediately when typed, not on enter, so that users know there is an issue ASAP ( like Sally said, it's just one less step) but either method works and this is not a major concern as long as they get feedback on the input eventually

I notice a couple of issues with the text copy, but those should be easy to fix! 

The copy text on the "Authentication" page has been changed to include the term "one-time-passcode", there should be an updated version in the COX Dashboard Figma. (let me know if you have any issues finding it)

Also, the login page has a typo (Login in), but otherwise everything looks great! Please let me know if you have any questions

**Review by geoerika — APPROVED** [review]

**Comment by hyx131** [comment]
> Looks great! For the error input field, I think it would be ideal to have the message appear immediately when typed, not on enter, so that users know there is an issue ASAP ( like Sally said, it's just one less step) but either method works and this is not a major concern as long as they get feedback on the input eventually
> 
> I notice a couple of issues with the text copy, but those should be easy to fix!
> 
> The copy text on the "Authentication" page has been changed to include the term "one-time-passcode", there should be an updated version in the COX Dashboard Figma. (let me know if you have any issues finding it)
> 
> Also, the login page has a typo (Login in), but otherwise everything looks great! Please let me know if you have any questions

LOL oh gawd that login typo 🤦‍♀️ , gonna fix

File: src/modules/login.js (hunk 1 above)
Comment (hyx131) [reply src/modules/login.js:160]: sounds good!

**Review by woozyking — APPROVED** [review]

**Pull request merged**